recipes/orc/all/conanfile.py

```diff
@@ -0,0 +1,149 @@
+    topics = ("orc", "columnar", "file-format", "hadoop")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_tools": [True, False],
+        "build_avx512": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_tools": False,
+        "build_avx512": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12",
+        }
+
+    @property
+    def _is_legacy_one_profile(self):
+        return not hasattr(self, "settings_build")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.compiler == "apple-clang":
+            # AVX support is not enabled by default, might need to add -mavx512f to CXXFLAGS
+            del self.options.build_avx512
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.12")
+        self.requires("lz4/1.9.4")
+        self.requires("snappy/1.1.9")
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("zstd/1.5.5")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src", build_folder="build")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def build_requirements(self):
+        if not self._is_legacy_one_profile:
+            self.tool_requires("protobuf/<host_version>")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def generate(self):
+        VirtualBuildEnv(self).generate()
+        VirtualRunEnv(self).generate(scope="build")
+
+        tc = CMakeToolchain(self)
+        tc.variables["ORC_PACKAGE_KIND"] = "conan"
+        tc.variables["BUILD_JAVA"] = False
+        tc.variables["BUILD_CPP_TESTS"] = False
+        tc.variables["BUILD_TOOLS"] = False
+        tc.variables["BUILD_LIBHDFSPP"] = False
+        tc.variables["BUILD_POSITION_INDEPENDENT_LIB"] = bool(self.options.get_safe("fPIC", True))
+        tc.variables["INSTALL_VENDORED_LIBS"] = False
+        tc.variables["BUILD_ENABLE_AVX512"] = False
```

Comment:
This conflicts with the `tc.cache_variables["BUILD_ENABLE_AVX512"] = self.options.get_safe("build_avx512", False)` below.

Also, can all of the `cache_variables` be turned into plain `variable`, maybe? Either way, you will have to add `tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"` for ORC versions below 1.9 because they use a CMake policy version less than 3.12 (https://github.com/apache/orc/blob/v1.8.6/CMakeLists.txt#L18).